src/Storage/Flysystem2Storage.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Kevinrob\GuzzleCache\Storage;
+
+use Kevinrob\GuzzleCache\CacheEntry;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FileNotFoundException;
+
+class Flysystem2Storage implements CacheStorageInterface
+{
+
+    /**
+     * @var Filesystem
+     */
+    protected $filesystem;
+
+    public function __construct(FilesystemAdapter $adapter)
+    {
+        $this->filesystem = new Filesystem($adapter);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function fetch($key)
+    {
+        if ($this->filesystem->fileExists($key)) {
+            // The file exist, read it!
+            $data = @unserialize(
+                $this->filesystem->read($key)
+            );
+
+            if ($data instanceof CacheEntry) {
+                return $data;
+            }
+        }
+
+        return;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save($key, CacheEntry $data)
+    {
+        return $this->filesystem->write($key, serialize($data));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
```

Comment:
Almost the same as above:

`\Kevinrob\GuzzleCache\Storage\CacheStorageInterface::delete` requires a `bool` return value, but `\League\Flysystem\Filesystem::delete` returns void, and throws exceptions. So you should add a `return true` after the ->delete call, and change the return the `catch` to `return false`